lib/random.php

```diff
@@ -198,67 +209,33 @@ function random_int($min, $max)
              * Provided by @CodesInChaos
              * 
              * @ref https://gist.github.com/CodesInChaos/03f9ea0b58e8b2b8d435
-             * 
-             * The rejection probability is at most 0.5, so this corresponds
-             * to a failure probability of 2^-128 for a working RNG
              */
-            for ($attempts = 0; $attempts < 128; $attempts++) {
-                /**
-                 * Generate a random integer...
-                 */
-                $rval = random_bytes(PHP_INT_SIZE);
-                if ($rval === false) {
-                    throw new Exception(
-                        'Random number generator failure'
-                    );
-                }
-                $value = 0;
-                for ($i = 0; $i < PHP_INT_SIZE; ++$i) {
-                    $value = ($value << 8) | ord($rval[$i]);
-                }
-                if ($value >= $min && $value <= $max) {
-                    /**
-                     * This value is within the accepted bounds.
-                     */
-                    return $value;
-                }
-            }
-            throw new Exception(
-                'random_int: RNG is broken - too many rejections'
-            );
-        }
-        /**
-         * We incremented $range earlier to test for overflows
-         */
-        --$range;
-
-        /**
-         * Initialize variables to 0
-         */
-        $attempts = $bits = $bytes = $mask = 0;
+            $bytes = PHP_INT_SIZE;
+            $mask = -1;
```

Comment:
To avoid using -1 (which assumes two's compliment) you could do ~0. But I don't think PHP runs on any machines that aren't two's compliment, and ~0 might even be hitting undefined behavior in the PHP C code (I haven't looked).